src/Fantomas.Core/CodePrinter.fs

```diff
@@ -300,14 +300,19 @@ let genInheritConstructor (ic: InheritConstructor) =
         genType node.Type
         +> sepSpaceOrIndentAndNlnIfExpressionExceedsPageWidth (genExpr node.Expr)
 
-let isSynExprLambdaOrIfThenElse (e: Expr) =
+let isSynExprIfThenElse (e: Expr) =
     match e with
-    | Expr.Lambda _
     | Expr.IfThen _
     | Expr.IfThenElif _
     | Expr.IfThenElse _ -> true
     | _ -> false
 
+let isSynExprLambdaOrIfThenElse (e: Expr) =
+    match e with
+    | Expr.Lambda _ -> true
+    | _ when isSynExprIfThenElse e -> true
```

Comment:
I really dislike the wildcard pattern with a when clause.
This should be its own partial active pattern.